packages/api/src/decorator/private/templateMiddleware.ts

```diff
@@ -1,20 +1,18 @@
 import { warnOnce } from 'botframework-webchat-core';
 import { createChainOfResponsibility, type ComponentMiddleware } from 'react-chain-of-responsibility';
 import { type EmptyObject } from 'type-fest';
-import { any, array, custom, safeParse, type Output } from 'valibot';
+import { any, array, function_, pipe, safeParse, type InferOutput } from 'valibot';
 
 export type MiddlewareWithInit<M extends ComponentMiddleware<any, any, any>, I> = (init: I) => ReturnType<M> | false;
 
 const EMPTY_ARRAY = Object.freeze([]);
 
-export default function templateMiddleware<Init, Request = any, Props extends {} = EmptyObject>(name: string) {
+export default function templateMiddleware<Init, Request = any, Props extends object = EmptyObject>(name: string) {
```

Comment:
Why `object` here? The `object` filters out only primitive values, but can be assigned an `Array` or function.

Reply:
ESLint rule: `{}` can be primitives.

`@types/react` say `type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>`

Will revert and ignore ESLint.